src/composables/useRoutesMap.ts

```diff
@@ -0,0 +1,227 @@
+      view.fit(extent, {
+        size: map.getSize(),
+        maxZoom: MAX_ZOOM_CENTERING_FACTOR,
+      });
+    }
+  };
+
+  /**
+   * Get the length of a route based on the geometry.
+   * @param {Feature | Ref<Feature>} route - The LineString route.
+   * @return {number} The length of the route.
+   */
+  const getRouteLength = (route: Feature | Ref<Feature>): number => {
+    const geom = unref(route).getGeometry();
+    if (geom instanceof LineString) {
+      return getLength(geom);
+    }
+    return 0;
+  };
+
+  /**
+   * Returns a string representing the length of a route in kilometers.
+   * @param {FeatureRoute} featureRoute - The route to calculate the length for.
+   * @return {string} A string representing the length of the route in kilometers.
+   */
+  const getRouteLengthLabel = (featureRoute: FeatureRoute): string => {
+    const length = getRouteLength(featureRoute.route);
+    return formatLength(length);
+  };
+
+  /**
+   * Formats the length value to a rounded string with the global route length unit.
+   * @param {number} length - The length value to format.
+   * @return {string} The formatted length string.
+   */
+  const formatLength = (length: number) => {
+    const lengthRoudned = Math.round(length * 100);
+    return `${lengthRoudned} ${i18n.global.t('global.routeLengthUnit')}`;
+  };
+
+  onMounted(() => {
+    useGeographic();
+  });
+
+  /**
+   * Centers the map on the current location if geolocation is supported.
+   * If geolocation is not supported, an error message is logged to the console.
+   * @return {void}
+   */
+  const centerOnCurrentLocation = (): void => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const coords = fromLonLat([
+            position.coords.longitude,
+            position.coords.latitude,
+          ]);
+          center.value = coords;
+          if (mapRef.value?.map) {
+            const view = mapRef.value.map.getView();
+            view.setCenter(coords);
+            view.setZoom(DEFAULT_MAP_ZOOM);
+          }
+        },
+        (error) => {
+          console.error('Geolocation error:', error);
+        },
+      );
+    } else {
+      console.error('Geolocation is not supported by this browser.');
+    }
+  };
+
+  /**
+   * Styling for the drawn routes.
+   * Uses function override to create styles for LineString vertices.
+   * @param feature Feature
+   * @returns {Style[]}
+   */
+  const styleFunction: OverrideStyleFunction = (feature) => {
+    const geometry = feature.getGeometry();
+    // basic styles for LineString
+    const styles = [
+      new Style({
+        stroke: new Stroke({
+          color: primaryColor,
+          width: 4,
+        }),
+      }),
+    ];
+    if (!geometry) return styles;
+    // style for mid-points
+    styles.push(
+      new Style({
+        geometry: new MultiPoint(geometry.getCoordinates()),
+        image: new Icon({
```

Comment:
Do not use emebedded SVG images.

Reply:
Replaced with import from /public assets.